### ISSUE TITLE
View/Edit Ports Via SNMP Poll fails on Juniper

### ISSUE THREAD

**Issue by pashdown** [inex/IXP-Manager#762]
##### ISSUE TYPE

Bug Report

##### OS

Ubuntu 20.04.3

##### VERSION

```
define( 'APPLICATION_VERSION', '6.2.0' );
define( 'APPLICATION_VERDATE', '2021111400' )
```

##### ENVIRONMENT 

```
# php -v
PHP 8.0.12 (cli) (built: Oct 22 2021 12:34:48) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.12, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.12, Copyright (c), by Zend Technologies
# dpkg -l | grep php
ii  libapache2-mod-php8.0                8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        server-side, HTML-embedded scripting language (Apache 2 module)
ii  php-common                           2:85+ubuntu20.04.1+deb.sury.org+1                                   all          Common files for PHP packages
ii  php-ds                               1.3.0-8+ubuntu20.04.1+deb.sury.org+3                                amd64        PHP extension providing efficient data structures for PHP 7
ii  php-igbinary                         3.2.6+2.0.8-2+ubuntu20.04.1+deb.sury.org+3                          amd64        igbinary PHP serializer
ii  php-memcache                         8.0+4.0.5.2+3.0.9~20170802.e702b5f9+-3+ubuntu20.04.1+deb.sury.org+3 amd64        memcache extension module for PHP
ii  php-memcached                        3.1.5+2.2.0-10+ubuntu20.04.1+deb.sury.org+3                         amd64        memcached extension module for PHP, uses libmemcached
ii  php-msgpack                          2.2.0~rc1+2.1.2+0.5.7-2+ubuntu20.04.1+deb.sury.org+3                amd64        PHP extension for interfacing with MessagePack
ii  php-rrd                              2.0.3+1.1.3-2+ubuntu20.04.1+deb.sury.org+3                          amd64        PHP bindings to rrd tool system
ii  php-yaml                             2.2.2+2.1.0+2.0.4+1.3.2-1+ubuntu20.04.1+deb.sury.org+3              amd64        YAML-1.1 parser and emitter for PHP
ii  php8.0                               8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               all          server-side, HTML-embedded scripting language (metapackage)
ii  php8.0-bcmath                        8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        Bcmath module for PHP
ii  php8.0-cgi                           8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        server-side, HTML-embedded scripting language (CGI binary)
ii  php8.0-cli                           8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        command-line interpreter for the PHP scripting language
ii  php8.0-common                        8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        documentation, examples and common module for PHP
ii  php8.0-curl                          8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        CURL module for PHP
ii  php8.0-ds                            1.3.0-8+ubuntu20.04.1+deb.sury.org+3                                amd64        PHP extension providing efficient data structures for PHP 7
ii  php8.0-gd                            8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        GD module for PHP
ii  php8.0-igbinary                      3.2.6+2.0.8-2+ubuntu20.04.1+deb.sury.org+3                          amd64        igbinary PHP serializer
ii  php8.0-intl                          8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        Internationalisation module for PHP
ii  php8.0-mbstring                      8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        MBSTRING module for PHP
ii  php8.0-memcache                      8.0+4.0.5.2+3.0.9~20170802.e702b5f9+-3+ubuntu20.04.1+deb.sury.org+3 amd64        memcache extension module for PHP
ii  php8.0-memcached                     3.1.5+2.2.0-10+ubuntu20.04.1+deb.sury.org+3                         amd64        memcached extension module for PHP, uses libmemcached
ii  php8.0-msgpack                       2.2.0~rc1+2.1.2+0.5.7-2+ubuntu20.04.1+deb.sury.org+3                amd64        PHP extension for interfacing with MessagePack
ii  php8.0-mysql                         8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        MySQL module for PHP
ii  php8.0-opcache                       8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        Zend OpCache module for PHP
ii  php8.0-readline                      8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        readline module for PHP
ii  php8.0-rrd                           2.0.3+1.1.3-2+ubuntu20.04.1+deb.sury.org+3                          amd64        PHP bindings to rrd tool system
ii  php8.0-snmp                          8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        SNMP module for PHP
ii  php8.0-xml                           8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        DOM, SimpleXML, XML, and XSL module for PHP
ii  php8.0-yaml                          2.2.2+2.1.0+2.0.4+1.3.2-1+ubuntu20.04.1+deb.sury.org+3              amd64        YAML-1.1 parser and emitter for PHP
ii  php8.0-zip                           8.0.12-1+ubuntu20.04.1+deb.sury.org+1                               amd64        Zip module for PHP
ii  php8.1-common                        8.1.0~rc5-2+ubuntu20.04.1+deb.sury.org+1                            amd64        documentation, examples and common module for PHP
ii  php8.1-ds                            1.3.0-8+ubuntu20.04.1+deb.sury.org+3                                amd64        PHP extension providing efficient data structures for PHP 7
ii  php8.1-igbinary                      3.2.6+2.0.8-2+ubuntu20.04.1+deb.sury.org+3                          amd64        igbinary PHP serializer
ii  php8.1-memcache                      8.0+4.0.5.2+3.0.9~20170802.e702b5f9+-3+ubuntu20.04.1+deb.sury.org+3 amd64        memcache extension module for PHP
ii  php8.1-memcached                     3.1.5+2.2.0-10+ubuntu20.04.1+deb.sury.org+3                         amd64        memcached extension module for PHP, uses libmemcached
ii  php8.1-msgpack                       2.2.0~rc1+2.1.2+0.5.7-2+ubuntu20.04.1+deb.sury.org+3                amd64        PHP extension for interfacing with MessagePack
ii  php8.1-rrd                           2.0.3+1.1.3-2+ubuntu20.04.1+deb.sury.org+3                          amd64        PHP bindings to rrd tool system
ii  php8.1-yaml                          2.2.2+2.1.0+2.0.4+1.3.2-1+ubuntu20.04.1+deb.sury.org+3              amd64        YAML-1.1 parser and emitter for PHP
```

##### CONFIGURATION

```
APP_DEBUG=false
APP_URL="https://ixp.slix.net"
APP_TIMEZONE="America/Denver"
APP_LOG="single"
IXP_FE_FRONTEND_CUSTOMER_ONE=customer
IXP_FE_FRONTEND_CUSTOMER_MANY=customers
IXP_FE_FRONTEND_CUSTOMER_OWNER="customer's"
IXP_FE_FRONTEND_CUSTOMER_OWNERS="customers'"
IXP_RESELLER_ENABLED=true
IXP_RESELLER_RESOLD_BILLING=true
IXP_AS112_UI_ACTIVE=true
IXP_FE_FRONTEND_DISABLED_LOGO=false
GRAPHER_BACKENDS="mrtg|smokeping"
GRAPHER_CACHE_ENABLED=true
GRAPHER_BACKEND_MRTG_DBTYPE="rrd"
GRAPHER_BACKEND_MRTG_WORKDIR="/srv/mrtg"
GRAPHER_BACKEND_MRTG_LOGDIR="/srv/mrtg"
GRAPHER_BACKEND_SMOKEPING_ENABLED=true
GRAPHER_BACKEND_SMOKEPING_URL="http://rc1.slix.net/smokeping"
IXP_API_JSONEXPORTSCHEMA_PUBLIC=true
VIEW_SKIN="slix"
CACHE_DRIVER=memcached
DOCTRINE_PROXY_AUTOGENERATE=false
DOCTRINE_CACHE=memcached
DOCTRINE_CACHE_NAMESPACE=IXPMANAGERNAMESPACE
IXP_IRRDB_BGPQ3_PATH=/usr/bin/bgpq3
AUTH_PEERINGDB_ENABLED=true
IXP_FE_FRONTEND_DISABLED_LOOKING_GLASS=false
```

##### STEPS TO REPRODUCE

- Switch
- Switch Ports
- View/Edit Ports (via SNMP poll)

##### EXPECTED RESULTS
To be able to edit and view my ports

##### ACTUAL RESULTS

"SNMP polling information failed or there are no Ethernet ports on this switch"

##### IMPORTANCE
I can't view edit ports.  I remember patching this for JunOS on 5.7.0, but I can't find what that patch was.  Would be nice if it just worked without patching.

**Comment by nickhilliard** [comment]
This may be fixed in 6d1dd152b2cc8a2d88fe87b3829056fade56d0eb.  If you're running 6.2.0 you should be able to run a git pull to this version.  Can you let us know if this fixes the problem?

**Comment by pashdown** [comment]
That did the trick!  Thanks Nick/Barry.
